### PR TITLE
VirtualHardDisk Download

### PR DESCRIPTION
**Pull request by damanMSFT**

### CODE REVIEW

File: rpc/common/moc_common_common.proto [hunk 1]
```diff
@@ -348,8 +349,9 @@ enum HyperVGeneration {
 } 
 
 enum DiskFileFormat {
-	DiskFileFormatVHDX =  0;
-	DiskFileFormatVHD = 1;
+	DiskFileFormatUNKNOWN = 0;
+	DiskFileFormatVHDX =  1;
+	DiskFileFormatVHD = 2;
```
Comment (msftpgayam) [inline rpc/common/moc_common_common.proto:354]: we should only append to the enum. If we change this order and an old object has DiskFileFormat based on the previous values, their meaning will change after the upgrade.

Comment (damanMSFT) [reply rpc/common/moc_common_common.proto:354]: Replied about this on teams

Comment (damanMSFT) [reply rpc/common/moc_common_common.proto]: Done

**Review by msftpgayam — DISMISSED** [review]

**Review by raghavendra-nataraj — APPROVED** [review]

**Review by msftpgayam — APPROVED** [review]

File: rpc/common/moc_common_common.proto [hunk 2]
```diff
@@ -350,6 +350,7 @@ enum HyperVGeneration {
 enum DiskFileFormat {
 	DiskFileFormatVHDX =  0;
 	DiskFileFormatVHD = 1;
+	DiskFileFormatUNKNOWN = 2;
```
Comment (liwonMS) [inline rpc/common/moc_common_common.proto:353]: Are we expecting cx to actually input Unkonwn. I couldn't really think of situation where this is needed. especially when default value for int32 is 0 (VHDX).

Comment (liwonMS) [reply rpc/common/moc_common_common.proto:353]: if we dont have cx flow that would allow this value to be passed down, This should be removed.

**Review by liwonMS — APPROVED** [review]

**Pull request merged**